### ISSUE TITLE
simple_uart_list lists only numeric UART Ports

### ISSUE THREAD

**Issue by andkae** [AndreRenaud/simple_uart#9]
Hi André

I use the _simple_uart_ for interfacing the [USB-ISS](http://www.robot-electronics.co.uk/htm/usb_iss_tech.htm) . In the Linux is this device recognized as ```/dev/ttyACM0```. With your filter will this device not listed.  How could look a bugfix? In my opinion would the best solution to prvide a filter in the function. What do you think?

https://github.com/AndreRenaud/simple_uart/blob/f932d00a6c0629d7ccd35270fd3ef7f81a5f8c3d/simple_uart.c#L447

Thanks for your support.

BR,
Andreas

**Comment by AndreRenaud** [comment]
Yes. You'll see that block is duplicated below for `ttyUSB*`. It should probably be refactored to look for all of `ttyS*`, `ttyUSB*` & `ttyACM*`. There may be others as well.

**Comment by andkae** [comment]
ok great, then i'll prepare a PR. Stay tuned ;)

**Comment by andkae** [comment]
as prototype for the new function
```c
int simple_uart_list(char ***namesp, char ***descriptionp, ***filterp)
```
if filterp is NULL we use '/ttyS[0-9]*' and ttyUSB[0-9]*, otherwise we use the filter array provided on filterp.

Is the approach suitable?

BR,
Andreas

**Comment by andkae** [comment]
in addition we can also overload the function via c macro to make it backward compatible

**Comment by akaeba** [comment]
please check #17 - it enables arbitrary port naming for Linux/Apple

**Comment by andkae** [comment]
solved in #18 . Thanks for adjusting.
